### PR TITLE
feat(extras): added gleam lang support

### PR DESCRIPTION
**Pull request by jluzny**
Added [Gleam lang](https://gleam.run/) support.

### CODE REVIEW

**Comment by rubiin** [comment]
you should also add lsp support as well.

**Comment by tkivela** [comment]
> you should also add lsp support as well. https://gleam.run/news/v0.21-introducing-the-gleam-language-server/ check here for more

Doesn't `nvim-lspconfig` configuration in the PR do effectively just that?

File: lua/lazyvim/plugins/extras/lang/gleam.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        vim.list_extend(opts.ensure_installed, { "gleam" })
+      end
+    end,
+  },
+
+  {
+    "neovim/nvim-lspconfig",
+    init = function()
+      require("lspconfig").gleam.setup({})
+    end,
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/gleam.lua:15]:
```suggestion
	opts = {
    	servers = {
    		gleam = {},
    	},
    },
```
This way the server will also get auto-installed by Mason.

Comment (dpetka2001) [reply lua/lazyvim/plugins/extras/lang/gleam.lua:15]: Oh it seems Mason has deprecated the installation for Gleam and it should be installed another way. Still, I believe the above changes make more sense in the spirit of how LazyVim configures LSP servers.

**Review by dpetka2001 — APPROVED** [review]

**Comment by rubiin** [comment]
> > you should also add lsp support as well. [gleam.run/news/v0.21-introducing-the-gleam-language-server](https://gleam.run/news/v0.21-introducing-the-gleam-language-server/) check here for more
> 
> Doesn't `nvim-lspconfig` configuration in the PR do effectively just that?

Oh yeah , my bad . probably missed that

File: lua/lazyvim/plugins/extras/lang/gleam.lua (hunk 1 above)
Comment (rubiin) [reply lua/lazyvim/plugins/extras/lang/gleam.lua:15]: True this looks sensible

**Review by rubiin — CHANGES_REQUESTED** [review]
Add mason install for the gleam language server
`
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        "gleam",
      })
    end,
  },
`

File: lua/lazyvim/plugins/extras/lang/gleam.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "gleam" })
+      end
+    end,
+  },
+
```
Comment (rubiin) [inline lua/lazyvim/plugins/extras/lang/gleam.lua:10]: {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        "elixir-ls",
      })
    end,
  },

File: lua/lazyvim/plugins/extras/lang/gleam.lua (hunk 1 above)
Comment (jluzny) [reply lua/lazyvim/plugins/extras/lang/gleam.lua:15]: Indeed, the installation of LSP via mason has been removed per the request of gleam's team; see https://github.com/mason-org/mason-registry/pull/3872

**Comment by dpetka2001** [comment]
@jluzny The lines with the `init` function is not needed at all (lines 13-15). See my [comment](https://github.com/LazyVim/LazyVim/pull/2802#discussion_r1544908976).

**Comment by folke** [comment]
Thanks!

**Pull request merged**